examples/keygen.c

```diff
@@ -0,0 +1,56 @@
+#include <stdio.h>
+#include <assert.h>
+
+#include "random.h"
+#include "secp256k1.h"
+
+
+void print_hex(unsigned char* data, size_t size) {
+    size_t i;
+    printf("0x");
+    for (i = 0; i < size; i++) {
+        printf("%02x", data[i]);
+    }
+    printf("\n");
+}
+
+int main(void) {
+    unsigned char seckey[32];
+    unsigned char compressed_pubkey[33];
+    unsigned char uncompressed_pubkey[65];
+    size_t len;
+    secp256k1_pubkey pubkey;
+    secp256k1_context* ctx = secp256k1_context_create(SECP256K1_CONTEXT_SIGN);
+    while (1) {
```

Comment:
I'd prefer not to loop here and instead bail out if we hit an invalid key.

 The probability of hitting an invalid key is negligible if your randomness is good.
So this loop is only relevant if your randomness is broken. And then it hides the fact that your randomness is broken.

Reply:
FWIW:
https://github.com/bitcoin/bitcoin/blob/master/src/key.cpp#L157
https://github.com/rust-bitcoin/rust-secp256k1/blob/master/src/key.rs#L118
https://github.com/bitcoin-core/secp256k1/blob/f39f99be0e6add959f534c03b93044cef066fe09/src/tests.c#L130:L141

Reply:
Hm, I see. I still think that my argument holds up. Maybe @sipa can chime in for the code in Core.

Reply:
While the probability of an invalid key is negligible (something like 1/2^128?),  I feel that if the verify function is provided in the library then it could still be informative to include in the examples.

```
if (!secp256k1_ec_seckey_verify(ctx, seckey)) {
	printf("Invalid secret key\n");
	return 1;
}
```